README.md

```diff
@@ -9,40 +9,61 @@ Jekyll plugin for building Jekyll sites with any public GitHub-hosted theme
+   remote_theme: benbalter/retlab
+   ```
+
 
 ## Declaring your theme
 
-Remote themes are specified by the `remote_theme` key in the site's config.
+Remote themes are specified by the `remote_theme` key in the site's config.  By default, this will use `https://github.com` as the host.  If you would like to specify your own host, either specify the host part of the `remote_theme` (e.g., `https://github.com/benbalter/retlab`) or by specifying `repository` (e.g., `https://github.com`).
 
-For public GitHub, remote themes must be in the form of `OWNER/REPOSITORY`, and must represent a public GitHub-hosted Jekyll theme. See [the Jekyll documentation](https://jekyllrb.com/docs/themes/) for more information on authoring a theme. Note that you do not need to upload the gem to RubyGems or include a `.gemspec` file.
+For GitHub, remote themes must be in the form of `OWNER/REPOSITORY`, and must represent a Jekyll theme. See [the Jekyll documentation](https://jekyllrb.com/docs/themes/) for more information on authoring a theme. Note that you do not need to upload the gem to RubyGems or include a `.gemspec` file.
 
 You may also optionally specify a branch, tag, or commit to use by appending an `@` and the Git ref (e.g., `benbalter/retlab@v1.0.0` or `benbalter/retlab@develop`). If you don't specify a Git ref, the `master` branch will be used.
 
-For Enterprise GitHub, remote themes must be in the form of `http[s]://GITHUBHOST.com/OWNER/REPOSITORY`, and must represent a public (non-private repository) GitHub-hosted Jekyll theme. Other than requiring the fully qualified domain name of the enterprise GitHub instance, this works exactly the same as the public usage.
+To use your own host, such as for Enterprise GitHub, you can specify `remote_theme` providing the full url (e.g., `https://GITHUBHOST.com/OWNER/REPOSITORY`), and must represent GitHub-hosted Jekyll theme. Alternatively, you can specify `repository`.  This works exactly the same as the GitHub usage.
+
+## Private and Internal Themes
+
+If you would like to use a private or internal hosted theme (or have other specific headers needs):
+
+1. Create a Personal Access Token following the [Creating a personal access token on GitHub guides](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+2. Add the `remote_header` in your `_config.yml`
+
+   ```yml
+   remote_header: 
+     Authorization: token <personal autorization token>
+   ```
+   Where the personal autorization token is the token  provided by the github repo setting page.
+
+   > :warning: **Storing credentials can lead to security issues** - As stated in the [Creating a personal access   token on GitHub guides](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) Treat your tokens like passwords and keep them secret. When working with the API, use tokens as environment variables instead of hardcoding them into your programs.
```

Comment:
Given that this is explicitly called out in the documentation, I'd be hesitant to create a workflow that requires committing the secret to the repository. Could we use an environmental variable as suggested?

Reply:
I fully agree that specifying an environment variable here would prevent "hardcoding" any credentials which can be a security hazard.  Unfortunately, unless some changes are done on the server side to allow the GitHub Pages to specify credentials for a privately hosted themes, this was the easiest way to solve the issue.

Reply:
So does this mean that there is no way for GitHub Secrets to be accessed in the GitHub Pages environment?